### PR TITLE
EOS-22007: post-disruptive upgrade fix and minor HA mini-provisioner refactoring

### PR DESCRIPTION
**Pull request by azheregelya**
## Problem Statement
<pre>
  <code>
  Story Ref (if any):
See commit msg.  </code>
</pre>
## Unit testing on RPM done
<pre>
  <code>
  No
  </code>
</pre>
## Problem Description
<pre>
  <code>
See commit msg.  </code>
</pre>
## Solution
<pre>
  <code>
See commit msg.  </code>
</pre>
## Unit Test Cases
<pre>
  <code>
No  </code>
</pre>

### CODE REVIEW

**Comment by indrajitzagade** [comment]
Please fix codacy issues.

File: conf/mini_provisioner/v2/setup.yaml [hunk 1]
```diff
@@ -37,6 +37,14 @@ ha:
     upgrade:
         cmd: /opt/seagate/cortx/ha/bin/ha_setup upgrade
         args: --config $URL
+        pre:
```
Comment (indrajitzagade) [inline conf/mini_provisioner/v2/setup.yaml:40]: These lines are coming from https://github.com/Seagate/cortx-ha/pull/471/

File: ha/setup/ha_setup.py [hunk 2]
```diff
@@ -791,6 +793,73 @@ def process(self):
...
+        try:
+            if os.environ['PRVSNR_MINI_LEVEL'] == 'cluster':
+                machine_id = self.get_machine_id()
+                cluster_id = Conf.get(self._index, f"server_node.{machine_id}.cluster_id")
+                mgmt_info: dict = self.get_mgmt_vip(machine_id, cluster_id)
+                node_count: int = len(self.get_nodelist(fetch_from=UpgradeCmd.HA_CONFSTORE))
+
+                Log.info(f"Performing post disruptive upgrade routines on cluster \
+                        level. cluster_id: {cluster_id}, mgmt_info: {mgmt_info}, node_count: {node_count}")
+                perform_post_upgrade(self.s3_instance, do_unstandby=False)
```
Comment (indrajitzagade) [inline ha/setup/ha_setup.py:858]: Still we are not passing mgmt_info and node_count?

File: ha/setup/ha_setup.py [hunk 3]
```diff
@@ -791,6 +793,73 @@ def process(self):
...
+        self.s3_instance = UpgradeCmd.get_s3_instance(machine_id)
+
+    def process(self):
+        """
+        Process command.
+        """
+        try:
+            if os.environ['PRVSNR_MINI_LEVEL'] == 'cluster':
+                machine_id = self.get_machine_id()
+                cluster_id = Conf.get(self._index, f"server_node.{machine_id}.cluster_id")
```
Comment (indrajitzagade) [inline ha/setup/ha_setup.py:852]: There was suggestion from architecture team not to assume that the same confstore as during the deployment will be provided during the runtime. We may need to store the required information in consul and use it during runtime.
cc @ajaysrivas

File: ha/setup/ha_setup.py (hunk 2 above)
Comment (azheregelya) [reply ha/setup/ha_setup.py:858]: Oops. You are right.

Comment (azheregelya) [reply ha/setup/ha_setup.py]: Fixed.

**Comment by indrajitzagade** [comment]
@azheregelya Please update the PR title.

File: ha/setup/ha_setup.py (hunk 3 above)
Comment (indrajitzagade) [reply ha/setup/ha_setup.py:852]: Checking with provisioner team.

Comment (indrajitzagade) [reply ha/setup/ha_setup.py:852]: From the chat with provisioner team, this seems to be fine.

**Review by indrajitzagade — APPROVED** [review]
Approving so that the PR could be merged and tested with a custom build.

**Review by ajaysrivas — APPROVED** [review]

**Pull request merged**